src/openforms/formio/components/custom.py

```diff
@@ -770,6 +771,156 @@ def as_json_schema(component: Component) -> JSONObject:
+        label=_("selected"),
+        allow_null=True,
+        help_text=_(
+            "Whether the child is selected by the user or not for further processing"
+        ),
+    )
+
+    def __init__(self, **kwargs):
+        self.component = kwargs.pop("component", None)
+        super().__init__(**kwargs)
+
+
+class ChildListField(serializers.Field):
+    def __init__(self, component, **kwargs):
+        self.component = component
+        super().__init__(**kwargs)
+
+    def to_internal_value(self, data):
+        if not isinstance(data, list):
+            raise serializers.ValidationError("Expected a list of children.")
+
+        serializer = ChildSerializer(
+            data=data,
+            many=True,
+            component=self.component,
+        )
+        serializer.is_valid(raise_exception=True)
+
+        validated = serializer.validated_data
+        self.validate_list(validated)
+        return validated
+
+    def to_representation(self, value):
+        return ChildSerializer(value, many=True, component=self.component).data
+
+    # TODO
+    # Add proper type hints when #2324 is completed (sync the frontend with the backend)
+    def validate_list(self, children):
+        component_key = self.component["key"]
+        submission = self.context["submission"]
+        prefill_data = submission.get_prefilled_data()
+
+        fm_immutable_variable = FormVariable.objects.filter(
+            source=FormVariableSources.user_defined,
+            prefill_plugin=FM_PLUGIN_IDENTIFIER,
+            prefill_options__mutable_data_form_variable=component_key,
+            form=submission.form.id,
+        ).first()
+
+        if fm_immutable_variable:
+            # we do not receive these fields from the frontend (since they are not used
+            # for now) so we have to exclude them from the data that needs validation.
+            initial_value = [
+                {
+                    key: (
+                        datetime.strptime(value, "%Y-%m-%d").date()
+                        if key == "dateOfBirth" and value
+                        else value
+                    )
+                    for key, value in child.items()
+                    if key
+                    not in (
+                        "dateOfBirthPrecision",
+                        "lastName",
+                        "affixes",
+                        "initials",
+                    )
+                }
+                for child in prefill_data[fm_immutable_variable.key]
+            ]
+
+            if not self.validate_selected_children(children):
+                raise serializers.ValidationError(_("Invalid data."))
+
+            # We also receive the boolean `selected` which is set on the frontend and
+            # therefore should not be part of the data for validation (in combination
+            # with the retrieved data from the external source)
+            for child in children:
+                child.pop("selected", None)
```

Comment:
I think it's a bit smelly that a `validate` function performs side effects by mutating its argument. To keep a pure function, it'd probably be best to `copy.deepcopy(children)`, otherwise weird things will happen if we will use the validation serializer output as data to save in variables (something we probably want to do in the long term)